src/library/scala/collection/generic/IsIterable.scala

```diff
@@ -127,6 +127,22 @@ object IsIterable extends IsIterableLowPriority {
       def apply(coll: CC0[A]): IterableOps[A, Iterable, C] = coll
     }
 
+  type Tupled[F[+_]] = { type Ap[X, Y] = F[(X, Y)] }
+
+  implicit def mapOpsIsIterable[K0, V0, CC0[X, Y] <: MapOps[X, Y, Tupled[Iterable]#Ap, CC0[X, Y]]]: IsIterable[CC0[K0, V0]] { type A = (K0, V0); type C = CC0[K0, V0] } =
+    new IsIterable[CC0[K0, V0]] {
+      type A = (K0, V0)
+      type C = CC0[K0, V0]
+      def apply(coll: CC0[K0, V0]): IterableOps[A, Iterable, C] = coll
+    }
+
+  implicit def mapViewIsIterable[K0, V0, CC0[X, Y] <: MapView[X, Y]]: IsIterable[CC0[K0, V0]] { type A = (K0, V0); type C = View[(K0, V0)] } =
+    new IsIterable[CC0[K0, V0]] {
+      type A = (K0, V0)
+      type C = View[(K0, V0)]
+      def apply(coll: CC0[K0, V0]): IterableOps[A, Iterable, View[(K0, V0)]] = coll
+    }
+
```

Comment:
We still need explicit `IsIterable` instances for maps, I assume due to `MapOps` having a different number of type parameters. I added the two instances here that should make existing tests pass. This is however incomplete, we also need instances for AnyRefMap, IntMap, LongMap.

So in the end we don't gain much by removing `IsMap` in terms of lines of code. In this case, have a preference to keep it.

@julienrf @SethTisue @szeiger what do you think?

Reply:
I think it’s simpler to keep `IsMap`. Otherwise, we can check carefully that all the previous `IsMap` instances now have a corresponding `IsIterable` instance.